aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabaseResponse;
+import software.amazon.awssdk.services.glue.model.GetDatabasesRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabasesResponse;
+import software.amazon.awssdk.services.glue.model.GetTablesRequest;
+import software.amazon.awssdk.services.glue.model.GetTablesResponse;
+import software.amazon.awssdk.services.glue.model.InvalidInputException;
+import software.amazon.awssdk.services.glue.model.UpdateDatabaseRequest;
+
+public class GlueCatalog extends BaseMetastoreCatalog implements Closeable, SupportsNamespaces {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GlueCatalog.class);
+
+  private final Configuration conf;
+  private final GlueClient glue;
+  // Catalog ID is the AWS account ID. When it is null, it uses the account of the Glue client.
+  private final String catalogId;
+  private final String warehousePath;
+
+  /**
+   * constructor with default glue client
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, String catalogId, String warehousePath) {
+    this(conf, GlueClient.create(), catalogId, warehousePath);
+  }
+
+  /**
+   * constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, GlueClient glue, String catalogId, String warehousePath) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.warehousePath = warehousePath;
+    validateWarehousePath();
+  }
+
+  private void validateWarehousePath() {
+    Preconditions.checkArgument(warehousePath != null && warehousePath.length() > 0,
+        "Glue warehouse path must be set");
+    Preconditions.checkArgument(warehousePath.charAt(warehousePath.length() - 1) != '/',
+        "Glue warehouse path %s must not end with /", warehousePath);
+  }
+
+  @Override
+  public void close() throws IOException {
+    glue.close();
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    String dbName = IcebergToGlueConverter.toDatabaseName(tableIdentifier.namespace());
+    String tableName = tableIdentifier.name();
+    return new GlueTableOperations(conf, glue, catalogId, dbName, tableName);
+  }
+
+  /**
+   * Use the same format as Hive for default warehouse location
```

Comment:
This java doc isn't clear to me. I think the summary here should describe what the method is doing "Returns the default warehouse location as stored in the glue service (?). Uses the same format as Hive" 

Then under return "default warehouse path as a string using Hive's formatting"

Does that sound right?

Reply:
Yeah let me just be precise with the behavior of it instead of saying it's the same as Hive.